### ISSUE TITLE
v6 - Domain counts are shown, but no individual domains shown.

### ISSUE THREAD

**Issue by rdwebdesign** [pi-hole/web#2739]
### Actual behavior / bug

[Domains (custom allow/block) not showing](https://discourse.pi-hole.net/t/domains-custom-allow-block-not-showing/65439) on the web interface

![image](https://github.com/pi-hole/web/assets/1385443/618d8c20-ed40-4276-a17b-4122ce089668)

Domain counts are shown, but no individual domains shown.

### Expected behavior

All domains should be displayed.

### Additional context

Users confirmed it is happening in Docker and bare metal installs.

**Comment by DL6ER** [comment]
FTL is sending the data correctly, the issue must be in the web interface code

**Comment by rdwebdesign** [comment]
> FTL is sending the data correctly

I'm not sure this is the case.

I added domains as a test. The domain is added (the domain count increases), but they still are not shown on the table.

The POST request is sent and the response returns the new item (every parameter is returned correctly), BUT the request `/api/domains` returns incomplete data:

```json
{
  "domains": [
    {
      "domain": "domain.com",
      "type": null,
      "kind": null,
      "comment": null,
      "groups": [ 0 ],
      "enabled": true,
      "id": 1,
      "date_added": 1697081542,
      "date_modified": 1697081542
    },
    {
      "domain": "domain2.com",
      "type": null,
      "kind": null,
      "comment": "comment",
      "groups": [ 0 ],
      "enabled": true,
      "id": 2,
      "date_added": 1697081745,
      "date_modified": 1697081745
    },
    {
      "domain": "domain3.com",
      "type": null,
      "kind": null,
      "comment": "comment",
      "groups": [ 0 ],
      "enabled": true,
      "id": 3,
      "date_added": 1697081810,
      "date_modified": 1697081810
    }
  ]
}

```

All these entries are `type=deny` and `kind=exact`, but the response for `/api/domains` returns `type` and `kind` as NULL.

If I try to filter by type using `/api/domains/deny`, the same response is shown (with nulls).

If I try `/api/domains/deny/exact`, type and kind are returned correctly (strange).

**Comment by DL6ER** [comment]
You are right, this is actually a regression of https://github.com/pi-hole/FTL/pull/1649.

The fix is very simple. It comes done to that FTL added the exact/deny (etc.) arguments only when it exactly knew what you were requesting (hence, your observation that `/api/domains/deny/exact` works). However, any wildcard requests (like everything or all regex, or all allowed entries, ...) returned `NULL`s because they have not been considered in the corresponding switch.

I will move this issue into FTL space and put up a bugfix for it.
